### ISSUE TITLE
ANSI: SIMILAR TO expressions are not parsed

### ISSUE THREAD

**Issue by PLBMR** [sqlfluff/sqlfluff#2730]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Followup to #2722 .

Though ANSI does support [`SIMILAR TO` expressions](https://www.cockroachlabs.com/docs/v20.2/sql-grammar.html#a_expr), `sqlfluff` cannot currently parse `SIMILAR TO` expressions in the `ANSI` dialect. For example, for `example.sql` specified below:

```sql
WITH cleaned_bear_financial_branch AS (
    SELECT
        branch_id,
        TO_NUMBER(CASE WHEN honey_numerical_code SIMILAR TO '[0-9]{0,7}.?[0-9]{0,2}' THEN honey_numerical_code ELSE NULL END, '24601') AS honey_numerical_code
    FROM bear_financial_branch
)

SELECT branch_id
FROM cleaned_bear_financial_branch
LIMIT 10;
```

We get the following parse error:

```bash
>>> sqlfluff lint --dialect=ansi example.sql 
== [example.sql] FAIL                                                                                                                                                             
L:   4 | P:  19 |  PRS | Line 4, Position 19: Found unparsable section: 'CASE
                       | WHEN honey_numerical_code SIMILAR T...'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect?
All Finished 📜 🎉!
```

It is important that ANSI is able to parse `SIMILAR TO` expressions so that we can cascade support for these expressions down to descendent dialects.

Thus, I think we should:
1. Refactor `SIMILAR TO` parsing in the `postgres` dialect into the `ansi` dialect
2. Adjust descendent dialects in a way so that this refactoring does not introduce bugs into any descendent dialects.

### Expected Behaviour

`sqlfluff` lints the `example.sql` above without parse errors.

### Observed Behaviour

`sqlfluff` lints the `example.sql` above with parse errors:

```bash
>>> sqlfluff lint --dialect=ansi example.sql 
== [example.sql] FAIL                                                                                                                                                             
L:   4 | P:  19 |  PRS | Line 4, Position 19: Found unparsable section: 'CASE
                       | WHEN honey_numerical_code SIMILAR T...'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect?
All Finished 📜 🎉!
```

### How to reproduce

1. Install requirements: `pip install dbt==0.20.0 sqlfluff==0.10.1 markupsafe==2.0.1`
2. create an `example.sql` file with contents specified above.
3. Run `sqlfluff lint --dialect=ansi example.sql`.

### Dialect

ansi

### Version

`sqlfluff==0.10.1`

### Configuration

`Python 3.8.2`

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
i'm not sure we should implement this in ANSI as many dialects do not support this. Seems like #2732 has implemented it in the main two dialects that do support it, and drives the implementation for any other dialect that needs it.
